### PR TITLE
#71: updates for platform-ui compatibility

### PR DESCRIPTION
**Pull request by samrichca**
related platform-UI PR https://github.com/icgc-argo/platform-ui/pull/2395

- add additional exports for missing platform-UI components
- fix some type errors
- remove babel module resolver because it wasn't working
- downgrade react 18 to 17 for platform-UI compatibility

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -14,16 +14,6 @@
   "plugins": [
     "@babel/plugin-transform-runtime",
     "@babel/plugin-proposal-optional-chaining",
-    "@babel/plugin-proposal-object-rest-spread",
-
-    [
-      "module-resolver",
```
Comment (samrichca) [inline babel.config.json:20]: this plugin isn't working so I removed it and updated all the `src` imports.

File: package.json [hunk 2]
```diff
@@ -73,15 +73,14 @@
     "@storybook/testing-library": "^0.0.13",
     "@svgr/webpack": "^6.3.1",
     "babel-loader": "^8.0.5",
-    "babel-plugin-module-resolver": "^3.2.0",
     "babel-preset-env": "^1.7.0",
     "husky": "^2.2.0",
     "plop": "^2.3.0",
     "prettier": "^2.0.5",
     "pretty-quick": "^2.0.1",
-    "react": "^18.2.0",
```
Comment (samrichca) [inline package.json:82]: downgraded react from 18 to 17 due to rendering issues

File: src/DropdownButton/index.tsx [hunk 3]
```diff
@@ -50,13 +50,18 @@ type DropdownButtonItemConfig<ValueType = string> = {
 export interface DownloadButtonProps<ValueType>
   extends Omit<React.ComponentProps<typeof Button>, 'onClick'> {
   onItemClick: (item: DropdownButtonItemConfig<ValueType>) => void;
-  onClick?: (e: React.SyntheticEvent<HTMLButtonElement, Event>, { toggleMenuOpen: boolean }) => any;
```
Comment (samrichca) [inline src/DropdownButton/index.tsx:53]: this type definition was incorrect and was renaming toggleMenuOpen to boolean

File: src/PageLayout/index.tsx [hunk 4]
```diff
@@ -72,13 +72,13 @@ export const PageBody = styled.div<PageBodyProps>`
     grid-template-columns: 1fr;
   }
 
-  & ${PageContent} {
+  & PageContent! {
```
Comment (samrichca) [inline src/PageLayout/index.tsx:75]: component selectors weren't working, changed to this syntax. good find @ciaranschutte

File: src/SubMenu/styledComponents.tsx [hunk 5]
```diff
@@ -56,6 +56,7 @@ const level1Style = ({ selected, theme }: StyleCalculationInput) => css`
   ${defaultLabelStyle({ theme, selected })}
   background: ${theme.colors.white};
   animation: all 1s;
+  width: 100%;
```
Comment (samrichca) [inline src/SubMenu/styledComponents.tsx:59]: submenus in File Repository facet panel weren't filling up the width of the column

File: src/ThemeProvider/index.tsx [hunk 6]
```diff
@@ -46,7 +44,6 @@ const ThemeProvider: React.ComponentType<{ theme?: keyof typeof themes }> = ({
 };
 
 export default ThemeProvider;
-export const useTheme = () => React.useContext(ThemeContext as React.Context<typeof defaultTheme>);
-export const styled: CreateStyled<typeof defaultTheme> = emotionStyled;
-
 export type Theme = typeof defaultTheme;
+export const useTheme = () => React.useContext(ThemeContext as React.Context<Theme>);
```
Comment (samrichca) [inline src/ThemeProvider/index.tsx:48]: use defined type instead of typeof

**Review by demariadaniel — APPROVED** [review]
Running + building smoothly lgtm 🚀

**Comment by ciaranschutte** [comment]
@samrichca what happened with the imports? Was the module resolution alias breaking something?

**Review by joneubank — APPROVED** [review]

**Pull request merged**